app/models/test_track/session.rb

```diff
@@ -16,6 +16,27 @@ def manage
     notify_unsynced_assignments! if sync_assignments?
   end
 
+  def visitor_dsl_for(identity)
+    if has_matching_identity?(identity)
+      visitor_dsl
```

Comment:
This fork is still a little ugly, because I didn't want to refactor too hard in this pass. Ultimately I'd like the session's primary identity to be just like any other identity we execute splits on.